project-base/storefront/components/Layout/Header/MenuIconic/MenuIconicItemLogin.tsx

```diff
@@ -0,0 +1,94 @@
+import { useTypedTranslationFunction } from 'hooks/typescript/useTypedTranslationFunction';
+import { MenuIconicItemLink, MenuIconicItemIcon, MenuIconicSubItemLink } from './MenuIconicElements';
+import { getInternationalizedStaticUrls } from 'helpers/localization/getInternationalizedStaticUrls';
+import { useDomainConfig } from 'hooks/useDomainConfig';
+import { useAuth } from 'hooks/auth/useAuth';
+import { Heading } from 'components/Basic/Heading/Heading';
+import { Popup } from 'components/Layout/Popup/Popup';
+import { Login } from 'components/Blocks/Popup/Login/Login';
+import { useState } from 'react';
+import { useCurrentUserData } from 'hooks/user/useCurrentUserData';
+import { ExtendedNextLink } from 'components/Basic/ExtendedNextLink/ExtendedNextLink';
+
+const TEST_IDENTIFIER = 'layout-header-menuiconic-login';
+
+export const MenuIconicItemLogin: FC = () => {
+    const t = useTypedTranslationFunction();
+    const { logout } = useAuth();
+    const { url } = useDomainConfig();
+    const { isUserLoggedIn } = useCurrentUserData();
+    const [customerUrl, customerOrdersUrl, customerEditProfileUrl] = getInternationalizedStaticUrls(
+        ['/customer', '/customer/orders', '/customer/edit-profile'],
+        url,
+    );
+    const [isLoginPopupOpened, setIsLoginPopupOpened] = useState(false);
+
+    const handleLogin = () => setIsLoginPopupOpened(true);
+
+    const MenuItem = isUserLoggedIn ? (
```

Comment:
Basically all HTML in this file depends on the login status. Would it make sense to actually split the templates into 2 components, maybe even load only the one which is displayed?

Reply:
I am not sure how much we can improve this. But I've created task for refactor this component. Since this is just reorganized code not refactored let's not make this task even longer.